src/lib/viewers/controls/page/__tests__/PageControlsForm-test.tsx

```diff
@@ -0,0 +1,168 @@
+            const wrapper = getWrapper({ onPageSubmit, pageCount, pageNumber });
+
+            getFormButton(wrapper).simulate('click');
+            getFormInput(wrapper).simulate('change', { target: { value: newPageNumber } });
+            getFormInput(wrapper).simulate('blur');
+
+            expect(onPageSubmit).toBeCalledWith(newPageNumber);
+        });
+
+        test('should not call onPageSubmit when input blurs with NaN', () => {
+            const newPageNumber = '';
+            const wrapper = getWrapper({ onPageSubmit, pageCount, pageNumber });
+
+            getFormButton(wrapper).simulate('click');
+            getFormInput(wrapper).simulate('change', { target: { value: newPageNumber } });
+            getFormInput(wrapper).simulate('blur');
+
+            expect(onPageSubmit).not.toHaveBeenCalled();
+        });
+
+        test('should handle when Enter key is pressed on input and new page number is valid', () => {
+            const key = ENTER;
+            const newPageNumber = 2;
+            const preventDefault = jest.fn();
+            const stopPropagation = jest.fn();
+            const wrapper = getWrapper({ onPageSubmit });
+            const initialFormButton = getFormButton(wrapper);
+            const initialFormInput = getFormInput(wrapper);
+
+            getFormButton(wrapper).simulate('click');
+            const intermediaryFormButton = getFormButton(wrapper);
+            const intermediaryFormInput = getFormInput(wrapper);
+            getFormInput(wrapper).simulate('change', {
+                preventDefault,
+                stopPropagation,
+                target: { value: newPageNumber.toString() },
+            });
+            getFormInput(wrapper).simulate('keydown', {
+                preventDefault,
+                stopPropagation,
+                key,
+                target: { value: newPageNumber.toString() },
+            });
+
+            expect(preventDefault).toHaveBeenCalled();
+            expect(stopPropagation).toHaveBeenCalled();
+            expect(onPageSubmit).toHaveBeenCalledWith(newPageNumber);
+            expect(initialFormButton.length).toBe(1);
+            expect(initialFormInput.length).toBe(0);
+            expect(intermediaryFormButton.length).toBe(0);
+            expect(intermediaryFormInput.length).toBe(1);
+            expect(getFormButton(wrapper).length).toBe(1);
+            expect(getFormInput(wrapper).length).toBe(0);
+        });
+
+        test('should handle when Enter key is pressed on input and new page number is invalid', () => {
+            const key = ENTER;
+            const newPageNumber = '';
+            const preventDefault = jest.fn();
+            const stopPropagation = jest.fn();
+            const wrapper = getWrapper({ onPageSubmit });
+            const initialFormButton = getFormButton(wrapper);
+            const initialFormInput = getFormInput(wrapper);
+
+            getFormButton(wrapper).simulate('click');
```

Comment:
It looks like there is a lot of duplication between these tests. Should we parameterize `newPageNumber` with test.each?